sympy/core/function.py

```diff
@@ -1871,6 +1871,39 @@ def __new__(cls, variables, expr):
         obj.nargs = FiniteSet(len(v))
         return obj
 
+    @classmethod
+    def from_func(cls, func):
+        """
+        'Lambda' object can also be created with a function.
+
+        A simple example:
```

Comment:
->
``` 
Examples
=======
```

Reply:
It seems like this can just be added to the instantiation of Lambda. You can detect a function with `if callable(foo)`.

Reply:
Thank you for your review!
Do you mean that I should modify `Lambda.__new__`?
I thought only 1 parameter is needed when using a `function` but 2 parameters are needed when using `variables` and `expr`. Will this cause confusion in future developing?

Reply:
Should I turn the instantiation of Lambda to this?
```
class Lambda(Expr):
    def __new__(cls, variables, expr = None):
        if callable(variables):
            from .symbol import Symbol
            variables = [Symbol(i) for i in func.__code__.co_varnames]
            expr = func(*variables)
        ......
```
Or this?
```
class Lambda(Expr):
    def __new__(cls, *args):
        if len(args)==1:
            from .symbol import Symbol
            variables = [Symbol(i) for i in func.__code__.co_varnames]
            expr = func(*variables)
        else:
            variables, expr = args
        ......
```